### PR TITLE
Artifact signature verification

### PR DESCRIPTION
**Pull request by tummychow**
WIP. Code to check signatures should be complete, code to actually perform signature checking is on the way.

### CODE REVIEW

File: pkg/hoist/hoist_launchable.go [hunk 1]
```diff
@@ -16,21 +16,24 @@ import (
...
-	Id            string  // A unique identifier for this launchable, used when creating runit services
-	RunAs         string  // The user to assume when launching the executable
-	ConfigDir     string  // The value for chpst -e. See http://smarden.org/runit/chpst.8.html
-	FetchToFile   Fetcher // Callback that downloads the file from the remote location.
-	RootDir       string  // The root directory of the launchable, containing N:N>=1 installs.
-	Chpst         string  // The path to chpst
-	Contain       string  // The path to contain
-	ContainerType string  // The container type to pass to contain
+	Location                string  // A URL where we can download the artifact from.
+	DigestLocation          string  // A URL to download the artifact's digest.
```
Comment (anthonybishopric) [inline pkg/hoist/hoist_launchable.go]: Don't forget to add docs to the wiki as needed.

File: pkg/hoist/verification_test.go [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+	"strings"
+	"testing"
+
+	. "github.com/anthonybishopric/gotcha"
+)
+
+func TestVerifySimpleTree(t *testing.T) {
+	workdir, err := ioutil.TempDir("", "verification")
+	Assert(t).IsNil(err, "temp dir error should be nil")
+	defer os.RemoveAll(workdir)
```
Comment (anthonybishopric) [inline pkg/hoist/verification_test.go]: uncomment?

File: pkg/pods/pod.go [hunk 3]
```diff
@@ -375,6 +377,63 @@ func (pod *Pod) Install(manifest *PodManifest) error {
...
+			}
+			defer fd.Close()
+
+			launchableDigest, err = digest.ParseDigest(fd)
+			if err != nil {
+				return err
+			}
+		} else {
+			digestSigPath := path.Join(temp, launchable.Version()+".sum.sig")
+			err = launchable.FetchToFile(stanza.DigestSignatureLocation, digestSigPath)
```
Comment (anthonybishopric) [inline pkg/pods/pod.go]: Leave a TODO here to extract the fetcher

**Comment by anthonybishopric** [comment]
looks really good :+1:

**Comment by tummychow** [comment]
Tested all four major cases in universe (correct digest, digest has extra file, digest has missing file, digest has mismatched sha) and they all returned the correct errors. Merging

**Pull request merged**